pio/lib/Sender/Sender.cpp

```diff
@@ -115,6 +115,35 @@ bool SenderClass::mqttConnect(const String &server, uint16_t port, const String
     return false;
 }
 
+#ifdef API_MQTT_HASSIO
+bool SenderClass::sendHassioDiscovery(String server, uint16_t port, String username, String password, String name, String unit)
+{
+    bool response = mqttConnect(server, port, name, username, password);
+    if (response)
+    {
+        _mqttClient.setBufferSize(512);
+        auto chipid = String(ESP.getChipId(), HEX);
+        String device = "\"dev\": { \"name\": \"" + name + "\",\"mdl\": \"ispindel\",\"sw\": \"" + FIRMWAREVERSION + "\",\"mf\": \"iSpindel\",\"ids\": [\"" + chipid + "\"]}";
+        String topic = "homeassistant/sensor/" + chipid;
+        _mqttClient.publish((topic + "_temperature/config").c_str(), ("{ \"uniq_id\": \"" + chipid + "_temp\", \"dev_cla\": \"temperature\", \"name\": \"Temperature\", \"unit_of_meas\": \"°" + unit + "\", \"val_tpl\": \"{{ value_json }}\", \"stat_t\": \"ispindel/" + name + "/temperature\"," + device + "}").c_str());
+        _mqttClient.loop();
+        _mqttClient.publish((topic + "_tilt/config").c_str(), ("{ \"uniq_id\": \"" + chipid + "_tilt\", \"name\": \"Tilt\", \"val_tpl\": \"{{ value_json }}\", \"stat_t\": \"ispindel/" + name + "/tilt\"," + device + "}").c_str());
+        _mqttClient.loop();
```

Comment:
For the sake of performance can the loop not be called eventually once and publish all together?

Reply:
Done.